proposals/infra/INF-0004-binding-resource-binding-diagnostics.md

```diff
@@ -0,0 +1,212 @@
+* Proposal: [0004](0004-binding-prefixes.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: TBD
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+*During the review process, add the following fields as needed:*
+
+* PRs: [#87578](https://github.com/llvm/llvm-project/pull/87578)
+* Issues: [#57886](https://github.com/llvm/llvm-project/issues/57886)
+
+## Introduction
+
+There's a specific set of valid types that can be bound as a resource in HLSL.
+When invalid types are bound to resources, helpful diagnostics need to be emitted
+so that the user knows why the bound type is invalid, or why the binding prefix

```

Comment:
> binding prefix

I assume this is the term used for the prefix to the register number, like `t0` or `u1`?  How well established is this?
Our [docs](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-variable-register) call this "register type".

I think that this introduction would be easier to follow if it had some examples and used them to explain the nomenclature that this document is going to be using.

Reply:
Ok, I've changed the terminology to use register type instead, and gave an example in the intro to define terms.

Reply:
Thank you - it's much easier to follow now!